[IMP] Migration to v8.0 and redesign.

nhomar opened the pull request:
## Resumen:
1. We have now a new "App" which install and put everything toghether.
2. Some major and minor pylint and pep8 errors fixed.
3. Normalization of imports alá v8.0.
4. View improvements using new button widgets on v8.0 (i.e.: [this view](https://docs.google.com/a/vauxoo.com/file/d/0B9dj_hHbN6p3czBRZ2ZhSGk0R0k/edit))
5. Migrated all data structure to v8.0 WMS.
6. Added some necesary default locations as data availables on `crm_rma_location_rma` y `crm_rma_advance_location` [see where](https://docs.google.com/a/vauxoo.com/file/d/0B9dj_hHbN6p3czBRZ2ZhSGk0R0k/edit)
7. Addapted tests and include a set of new tests for migrated modules.

**module**: `product_warranty`
1. Migrated to v8.0
2. New field `crm_return_address_id` to define the address to manage the returns.
3. Unit tests.

**module**: `crm_rma_lot_mass_return`
1. Migrated to v8.0
2. Wizard metasearch, in order to search in one place for several products to improve the speed of the data loading [see it in action](https://youtu.be/7DV8bJN9q88)

**module**: `crm_claim_rma`
1. Migrated to v8.0
2. to configure claim type in stage [see it in action](https://www.youtube.com/watch?v=oyT28yKWHDo&feature=youtu.be).
3. The method _warranty_limit_values is merged with method set_warranty_limit. (comment in spanish: el _set_warranty_limit.

Some major added fields also.

This PR will be squashed.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oca/rma/26)

<!-- Reviewable:end -->

gurneyalex commented:
CLA fixed

gurneyalex commented:
needs rebasing

nhomar commented:
@gurneyalex rebase was merged on our side to validate all is ok, now it is mergeable.

Can you give your +1 in order to have this last state on main branch.

Regards.

gurneyalex commented:
@nhomar one last flake8 issue to handle:

```
./crm_rma_lot_mass_return/wizard/returned_lines_from_serial.py:286:43: E128 continuation line under-indented for visual indent
```

nhomar commented:
@gurneyalex pylint fixed.

I think we should merge this as is, we have something like 20 more tasks related to rma and in the middle it becomes huge and huge it will be difficult to review and test by community.

Regards.

pedrobaeza commented on .travis.yml:
```diff
@@ -10,16 +13,18 @@ addons:
   - VERSION="8.0" LINT_CHECK="1"
-  - VERSION="8.0" ODOO_REPO="odoo/odoo" LINT_CHECK="0"
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" LINT_CHECK="0"
+  - VERSION="8.0" ODOO_REPO="odoo/odoo" LINT_CHECK="0" INCLUDE="rma"
```
Why restricting to only this module?

pedrobaeza commented on crm_claim_rma/crm_claim_rma.py:
```diff
@@ -0,0 +1,608 @@
+class crm_claim(models.Model):
+    _inherit = 'crm.claim'
+
+    def init(self, cr):
```
You should depend on crm_claim_code module (https://github.com/OCA/crm/tree/8.0/crm_claim_code) that do this better

yaniaular replied:
@pedrobaeza I think your proposal good about ´ You should depend on crm_claim_code module (https://github.com/OCA/crm/tree/8.0/crm_claim_code) that do this better ´. I'll have RMA depends on that module.

pedrobaeza commented:
@nhomar, I make the same comment as to Aristóbulo: you should split the PR into smaller ones with one module per PR.

pedrobaeza commented:
In an overview I have made, you are not using CamelCase for class names, so this style convention should be also applied.

nhomar commented:
@pedrobaeza 

About this

> >  @nhomar, I make the same comment as to Aristóbulo: you should split the PR into smaller ones with one module per PR.

I am sorry but it is a huge project on our side with a huge scope in this case particularly I can not include in our internal process make PR with module per module (too expesive in terms of time for us, and I think it do not add any value to the project as you can see a HUGE number of unit testing has been included to avoid the most hidden mistakes as we can) as this was an almost virgin repository in v8.0 we made the work, put public some plans and as you can see it is the repository we are using and validating with our customers, I can not split this specific job.

About this:

> > Why restricting to only this module?

This module is and "App Module" that test the integration of all validated modules, then even if you can use one by one it make tests very efficient saving .travis time, and testing the integration of all the repository.

We do not need to click "install" one over other as the .travis actually does for an integrated repository and functional situation like this.

About this:

> > In an overview I have made, you are not using CamelCase for class names, so this style convention should be also applied.

I honestly will not dedicate time to this on this specific PR (and BTW I think we should have a pylint rule for this if it is mandatory which I think it is not yet).

We can invest after and put in TODO such job. ( If you have some script that do that I can run it here ) but I think it is a low priority issuel, can you give your +1 without that?

About this:

> > You should depend on crm_claim_code module (https://github.com/OCA/crm/tree/8.0/crm_claim_code) that do this better

Can you extend a little please?

In the case we depend of that module I think it should be moved here, the name look like an rma (which is equal to claim in several cases), don't you think?

Thanks for the feedback!!!

pedrobaeza commented:
Nhomar, sorry but reviewing this PR is going to be impossible with this amount of modules/code. Making separated PRs with one module on each is independently from your overall project itself.

nhomar commented:
Ok. @pedrobaeza 

Can we wait for other PoV? I think we should submit this to vote.

IMHO it is (and I am sorry for the hard word) almost stupid allow a NOT tested repository new and stop a full migration for that reason.

And BTW the project that finance the Open Scopes "Can not be slighted" just because some silly technical rules.

As you can see (not always you can make only code per review) you can simply download -i rma and magically everything is running, and you can test functionally.....

Sorry but this time I disagree with you.

Regards.

nhomar commented:
Ah! another point, lose all the commit history is more dangerous in my opinion that simply merge it and wait for open issues.... dont' you think? how do I redo 3 month of work ? I do not understand now your position I am sorry.

pedrobaeza commented:
Use git filter-branch like this: https://github.com/odoomrp/odoomrp-wip/wiki/Migration-to-OCA

mistotebe commented:
@nhomar, is there a reason why you ignored my request for help in #20 (resulting in #24) and started your own afterwards? Some communication would have saved both of us a lot of time and effort. Let's look into merging the work in both to make a better one.

mistotebe commented:
OK, I can see that you might have started porting a week or two before I did and took the entire repo by storm regardless of what was happening to it in the meantime.

Maybe there is a missing process with OCA repos where it might be a good idea to announce one's intent to start porting in an issue against the repo to track the progress and reduce duplication of effort? @pedrobaeza and other maintainers, what are your thoughts on documenting this in a `CONTRIBUTING` document in the OCA repos?

pedrobaeza commented:
Yeah, this is a recurrent problem that people overlap efforts without announcing, and then both don't want to lose their part of the work. We must improved contributing documentation about this part indeed. Now, we need to let egos apart and try to solve this situation. Maybe a little code sprint? I'm willing to be present at it to help you to unlock this.

nhomar commented:
Hello @mistotebe

The issue is that when our customers ask us finance this migration, we didn't know about this comments https://github.com/OCA/rma/issues/20#issuecomment-126907139  (sorry my bad I didn't see  it before), then we tried to migrate everythign without touch the actual functionality (between april and may) and some stuff was not approved by customer analiysis.

At the end we re-do everything on new api (trying to mantain megeable compatibility).

You will be able to see the use cases in the list of tests and let me translate from spanish the user Stories in user side.

And then here we are 3 iterations done by us in our repository to fix everything with the bless of the end user already.

And 2 jobs re-done one here and one on ezee's side.

If you look in our code (which I know can have a lot of details internally but is already migrated to new api and with a bunch of unit tests) we tried to finish the migration of 100% of the repository to new api and new WMS.

And as our customer asked us link everything with Logistic, we added such cases also.

That's our actual situation, sadly I had short deadlines and we tried to make everything mergeable and tested to manage only 1 public effort in our side to join the job to the OCA, I can not as you see now expend X more months waiting to have them mergeable, we (vauxoo) sadly have 2 options:
1. Use as it is to move on in the next 20 or 30 features we need to develop.
2. Make a first merge as it is in order to start the process of officialize some job and use the "stable version" in our side.

Why I am affronting this situation?

Because even if we split the job we use a "blessed" version of everything and we are too close to deploy in production, one time we deploy in production I can not affront the job to migrate afterwards from Vauxoo's version to OCA's blessed version, on this point one time we (vauxoo) decide what to do on this moment that will be the version we will use/promote.

## Why am I so so insistent on that OCA use our version and not others?.

Let me share with you how is the customer (not names simply the use case).

This customer is a huge Electronic Seller with a huge amout of Parts Customer claimes, usage of serials, AND everything related with 3 internal process Supplier warranty, Its own warranty and QA and repairs.

As you can see it will be really difficult to find a customer with more cases of study where is open to share every single effort to public releases.

Then falling appart of Ego's (I know all this explanation can be taken as we are egomaniacs).

We do not want you to use our code because we want to have this development available as an official OCA repository because it is not a "little" improvement it is a revolutionary way to manage RMA in Odoo's world.

Even as you can see we didn't add our commercial references yet to the code (and it does not matter now to us) we really need to share this and start the open discussion with this as starting point.

Let me say the next steps (merged or not):
1. Clean up more the code:  Apply CamelCase, (improve pylint on mqt for that),
2. Increase the number of use cases on tests.
3. Create a full Supplier RMA system integrated with this one.
4. Reports.
5. Installer.
6. Full documentation.
7. More unit tests.
8. And at least and more important release in production.
9. Maintain in production.
10. Fix bugs properly.
11. Maintain public repository (OCA or NOT).
12. Redesign some website_\* interfaces to automate some process (like ask for a claim related to this flow throught the site).

Then as you can see we can not expend the short budget we have making this first PR as clean as @pedrobaeza is asking to do because the work that it needs now is more important that the burocracy stuff.

You can test this PR (actually is green) and it will be impossible (partially done or not) test with only code per review, we really ask community to test it using it and giving the clicks to ensure all works as expected.

How to test it:
1. Use runbot. https://runbot.odoo-community.org/runbot/repo/github-com-oca-rma-145
2. Download and install the repository (as you see in our internal production ones):
   `shell
     git clone https://github.com/OCA/rma -b 8.0
   `
3. Read functions and functionality (for example the comment done by @pedrobaeza about re-write a little functionality to use another oca repository will be done absolutlly).

But then How do I see the situation?

Now we have a v8 with 1 module and a v7 red,

After this PR and the little fixes we mentioned we will do we will have a Full tested v8.0 working and with a real case using all modules mixed and after all green.

Don't you think is unfear ask somebody else finance a job that really do not add now any value (it do not mean in a future we will make all more clean because we have an strongest starting point) but now don't you think is really egoist ask for such job?

I really can not believe nobody else thinks like me, it is not a matter of politics it is a matter of efficiency, if we ask all partners that want work open and improve the repositories that are DEAD like this one put them alive and continue using them internal repositories (because the starting point is not done part by part), What is the advange for them of do such contribution then?

As I said, the important parts will be done:
1. CamelCase with improved pylint job mentioned by @pedrobaeza.
2. depend in functionality of the other module  `crm_claim_code` he mentioned in comment.

We want maintain OCA's repository with our day by day job, not the Vauxoo's ones.

Regards.

gurneyalex commented:
please add Odoo Community Association (OCA) to authors list in `__openerp__.py` where relevant. 

and @nhomar frankly this set of modules may be the best things invented since sliced bread, but you're not doing yourself a favor, and you're not helping reviewers by proposing a PR affecting 217 files, completely removing some modules and not explaining the migration path.

nhomar commented:
> > and @nhomar frankly this set of modules may be the best things invented since sliced bread, but you're not doing yourself a favor, and you're not helping reviewers by proposing a PR affecting 217 files, completely removing some modules and not explaining the migration path. 

Hello @gurneyalex you are right about the roadmap, as I said before we will rebase this PR in order to allow us to list the changes and each explanation.

About the quantity of files, that's the point one by one this set of modules will be broke, because they are 17 dependencies each other and tests, IMHO it is easy test this PR as only one PR than separate all the job per module.

We can expend easily 1 more month fixing the details around that, but please dont look the diff (that will not say anything) it must be reviewed as "features".

That's like the new migration of accounting or the new WMS has been asked be migrated module per module by al one time it received 6 month of work, it was rebased and tested separatelly, and then merged as one (too much collateral effects).

And this PR particularly is bringing alive a full repository, as you can see only 1 module were migrated before this.

I want help you, but you need to help us also on be a little more flexible.

PS: We will continue rebasing and working on this, we found some ways to clean up a little the commit history to help you to check the path followed.

And the documentation will be included also in order to help you understand the change.

Thanks for the feedback.

nhomar commented:
@yaniaular Please verify that this PR https://github.com/OCA/rma/pull/32 can work with your job, IMHO that one is more complete in terms of migration to new api.

Then make a cherrypick to allow one time that PR is merged this PR is compatible yet.

nhomar commented:
@yaniaular Try this one also.

https://github.com/OCA/rma/pull/24

Remember that the coverage of your tests should increase in order to avoid the coverage all read.

yaniaular commented:
PR for product warranty module is splitted: https://github.com/OCA/rma/pull/33/files

dreispt commented:
@gurneyalex AFAIK module README files can be rst, md or txt. They are usually RST because the text is copy-pasted from the manifest description key, and there they are ususally in RST.

@nhomar https://twitter.com/stuk/status/625790198631170048

gurneyalex commented:
@dreispt ah and then _you_ would know that... https://github.com/odoo/odoo/commit/ec0b770ed3c4f4fbdf0f24930aaa05bcc5cb0b53 
Sorry I was pretty sure only rst was supported, and I don't know where I got that idea from...

nhomar commented:
We will make all the split on modules (not so happy but good no problem). 

We will left this PR open to record the progress of the PR's one time all is done I personally will close this issue with all the historical of the migration.

pedrobaeza commented:
That's true that Odoo load any rst, md or txt, but the parser is only rst, so at the end, you only have RST, not markdown.

nhomar commented:
https://github.com/OCA/rma/pull/33 Ready for product_warranty.

nhomar commented:
https://github.com/OCA/rma/pull/32 will be user for crm_claim_rma just little fixes needed there.

Commit:
[REF] crm_claim_type: add module version

Commit:
[REF] crm_claim_type: correct installation, usage, bug tracker link and rst format

Commit:
[REF] crm_claim_type: replace help for claim_type field

Commit:
[REF] crm_claim_type: set translatable name and description

Commit:
[REF] crm_claim_type: change spanish term for claim

Commit:
[REF] crm_claim_type: change group id for security access rule

Commit:
[REF] crm_claim_type: fix readme bugtracker link

Commit:
[REF] crm_claim_type: fix readme bugtracker link

Commit:
[REF] crm_claim_type: fix readme bugtracker link

Commit:
Merge branch '8.0-oca-corrections-for-crm_claim_type-dev-osval' of github.com:vauxoo-dev/crm into 8.0-oca-corrections-for-crm_claim_type-dev-osval

Commit:
[REF] crm_claim_type: correct spanish terms and add new ones for es_MX and es_VE

Commit:
[ADD] crm_claim_type: inherit crm claim form view

Commit:
[FIX] crm_claim_type: add permissions both groups technical features and salesman

Commit:
[REF] crm_claim_type: add license badget and new button, and remove useless line from Usage section

Commit:
[REF] crm_claim_type: correct typos in terms

Commit:
[REF] crm_claim_type: update license header agreements

Commit:
[REF] crm_claim_type: update readme file

Commit:
[REF] The demo of crm_claim has claim_type like selection field, and in this module is changed to many2one field, then the value of claim_type in the demo data changes

Commit:
[REF] guidelines applied in view

Commit:
[FIX] bad name in __OpenERP__.py about xml file

Commit:
[ADD] crm_claim_type: add crm_claim_stage based on crm_claim_type

crm_claim_stage functionality was moved from crm_claim_rma to crm_claim_type to
fulfill crm_claim_type module function, also aditional data and demo was added

Commit:
[FIX] crm_claim_type: remove already deleted file entry from __openerp__

Commit:
[REF] cmr_claim_type: add missing terms to i18n

Commit:
[REF] crm_claim_type: add visual logic and switch field positions

Commit:
[REF] crm_claim_type: remove redundant field stage_name_uniq from module

Commit:
[REF] crm_claim_type: add group to claim_type

Commit:
[REF] crm_claim_type: extend module description in README file

Commit:
[REF] crm_claim_type: move claim_type field

Commit:
[REF] crm_claim_type: unset translatable to claim_type

Commit:
[REF] crm_claim_type: add demos, override extending data for crm_claim

Commit:
[REF] crm_claim_type: add EOF

Commit:
[REF] crm_claim_type: update i18n terms

Commit:
[ADD] RMA updated with squash proposed to OCA/rma

nhomar committed:
Merge pull request #44 from vauxoo-dev/8.0-rebase_improvements_in_squash_branches

8.0 rebase improvements in squash branches

Commit:
[FIX] return statement in finally block may swallow exception corrected and Invalid attribute name with CamelCase corrected

Commit:
[FIX] when date, warehouse_id or claim_type is changed the lines were deleted because claim_lines empty was assigned

Commit:
[IMP] when the picking is carrier_loss the picking_type must be out

Commit:
[IMP] rebase with squash second part

Commit:
[IMP] delivery_address_id field is repeated in crm.claim view

nhomar committed:
Merge pull request #47 from vauxoo-dev/8.0_delivery_address_id_field_repeated-dev-yani

[FIX] delivery_address_id field is repeated in crm.claim view

hugho-ad committed:
[FIX] recover all moves for all picking_ids that are comming en the SO

hugho-ad committed:
[REF] change variable name  and remove unnecessary line

hugho-ad committed:
[FIX] inv_line.move_ids by inv_line.move_id

Commit:
Merge pull request #48 from vauxoo-dev/8.0-FIX-crm_rma_prodlot_invoice-dev-hugho-ad

[FIX] recover all moves for all picking_ids that are comming en the SO

Commit:
[MERGE] with 8.0

Commit:
[ADD] improvements from squashes

nhomar committed:
Merge pull request #46 from vauxoo-dev/8.0-rebase_improvements_in_squash_branches_2

[IMP] rebase with squash second part

Commit:
[FIX] cycle in api.multi method

Commit:
[IMP] improvements in receive products usign locations of rma and carrier_loss, guidelines applied

moylop260 committed:
[REF] .travis.yml: Change to MQT of vauxoo

nhomar committed:
Merge pull request #49 from vauxoo-dev/8.0_advance_location_updated_guidelines_and_locations-dev-yani

[IMP] improvements in receive products usign locations of rma and car…

KarenKawaii committed:
[REF] .travis.yml: Standardize file changing tests and adding matrix.

Commit:
Merge pull request #55 from vauxoo-dev/8.0-standardize-travis-file-rma-dev-karenkawaii

[REF] .travis.yml: Standardize file changing tests and adding matrix.

Commit:
[DEL] commented lines

Commit:
Merge pull request #56 from vauxoo-dev/8.0_delete_commented_lines-dev-yani

[DEL] commented lines

Commit:
[IMP] update from squashs

Commit:
[ADD] stock logistic workflow added to depends

Commit:
[IMP] update from squash

nhomar committed:
Merge pull request #57 from vauxoo-dev/8.0-updated_rma_repo_with_squash-yani

[ADD] update with squash part third

Commit:
[FIX] always the serial number must be unique and For the two pickings (pack and ship) must be used the same serial number and The picking call force_assing method for work. fix #60 #61

Commit:
Merge pull request #62 from vauxoo-dev/8.0-correcting_tests_rma_modules-dev-yani

[FIX] always the serial number must be unique and For the two picking…

Commit:
[FIX] return browse of claim line and result_2 is not define

Commit:
[ADD] alert.js was deleted by mistake. Fix #63

nhomar committed:
Merge pull request #64 from vauxoo-dev/8.0_fix_alert_js_deleted_by_mistake-dev-yani

[ADD] alert.js was deleted by mistake. Fix #63

Commit:
[IMP] update repo with squash number 4

Commit:
[IMP] jslint fixed and tests fixed

nhomar committed:
Merge pull request #65 from vauxoo-dev/8.0_update_rma_repo_with_squash_3

[IMP] update repo with squash number 4

Commit:
[IMP] update rma repo with squash part 5th

nhomar committed:
Merge pull request #66 from vauxoo-dev/8.0-rebase_squashs_five

8.0 rebase squashs five

Commit:
[IMP] update_squash_six

Commit:
[IMP] translations on crm_claim_rma module

nhomar committed:
Merge pull request #67 from vauxoo-dev/8.0-update_squash_six

8.0 update squash six

Commit:
[FIX] Send a message had a error and now is fixed

Commit:
[IMP] tree view of Stock RMA just can display products with qty > 0

Commit:
[IMP] translations on crm_rma_stock_location module

Commit:
[IMP] translations on crm_rma_stock_location module

Commit:
Merge branch '8.0' of github.com:Vauxoo/rma into 8.0

Commit:
[IMP] eval by safe_eval, load product with supplier claim, customer claim and others

Commit:
[IMP] Readme improved and tests added

nhomar committed:
Merge pull request #68 from vauxoo-dev/8.0-improvements_to_load_products_and_from_squash-dev-yani

Improvements on modules of crm_rma_advance_location, crm_rma_lot_mass_return and crm_rma_stock_location

Commit:
[IMP] allow invoice without lot number

Commit:
[IMP] avoid products repeated and allow supplier invoices without serial

moylop260 committed:
[REF] crm_claim_rma: Fix 'UnicodeWarning' removing special character 'n°'

Fix `UnicodeWarning: Unicode unequal comparison failed to convert both arguments to Unicode - interpreting them as being unequal`

Commit:
[IMP] take in account the products are in a same invoice line and separate to  differentiate, also, check the type of claim

Commit:
[ADD] Test constrains

Commit:
[IMP] ignore items repeated

Commit:
[FIX] invoice line with products with lot and other without lots

Commit:
[ADD] add demo data about purchase, invoice, picking, transfer, lot and sales

Commit:
[ADD] two claims more for test

Commit:
[REF] refactor in demo data

nhomar committed:
Merge pull request #69 from vauxoo-dev/8.0-allow_invoice_without_lot_number-dev-yani

[IMP] allow invoice without lot number

Commit:
[FIX] message in wizard is fixed because spelling mistake

Commit:
[FIX] message in wizard is fixed because spelling mistake

nhomar committed:
Merge pull request #70 from vauxoo-dev/8.0-fix_message_wizard-dev-yani

[FIX] message in wizard is fixed because spelling mistake

Commit:
Merge branch '8.0' into 8.0-fix_message_wizard-dev-yani

Conflicts:
	crm_rma_lot_mass_return/i18n/crm_rma_lot_mass_return.pot

Commit:
Merge pull request #71 from vauxoo-dev/8.0-fix_message_wizard-dev-yani

8.0 fix message wizard dev yani

Commit:
[REF] crm_rma_location_rma -> crm_rma_location

Commit:
Merge pull request #72 from vauxoo-dev/8.0-update-crm_rma_location-osval

crm_rma_location_rma -> crm_rma_location

Commit:
[FIX] crm_rma_lot_mass_return: fix issue #73

Commit:
[REF] crm_rma_lot_mass_return: remove comment, add test to challenge claim line description and name set

Commit:
[REF] crm_rma_lot_mass_return: test the messaged noticing a lot is not going to be added, remove comments, re-indent code for less lines

Commit:
[FIX] location input is wrong for this module, because pick and ship is not used, the products purchased from suppliers to stock only

Commit:
Merge pull request #75 from vauxoo-dev/8.0-fix_quantity_in_wrong_location-dev-yani

[FIX] location input is wrong for this module, because pick and ship …

Commit:
[REF] crm_rma_lot_mass_return: assert that claim lines keep the same after try to add a non-valid serial/lot number

Commit:
[IMP] crm_rma_lot_mass_return: add much more simpler claim_origin value assertion instead of for loop

moylop260 committed:
Merge pull request #74 from vauxoo-dev/8.0-fix-crm_rma_lot_mass_return-issue-#73-osval

8.0 fix crm_rma_lot_mass_return #73

JulioSerna committed:
[IMP][crm_rma_prolodlot_supplier] the context can be propage from another models and needs to validate model in context

JulioSerna committed:
[IMP][pep8]

Commit:
Merge pull request #76 from vauxoo-dev/8.0-rma-crm_rma_prodlot_sipplier-context-dev-julio

[IMP][crm_rma_prolodlot_supplier] the context can be propage from ano…

Commit:
[REF] crm_claim_rma: add changes applied to OCA into crm_claim_rma

Commit:
[FIX] crm_rma_advance_warranty: change get_destination_location is called

Commit:
[REF] crm_claim_rma: add readonly attr to claim_type

Commit:
Merge pull request #78 from vauxoo-dev/8.0-update-crm_claim_rma-with-minor-changes-osval

[REF] crm_claim_rma: applied OCA changes into crm_claim_rma

Commit:
[REF] crm_rma_lot_mass_return: change i18n translations and current components strings

Commit:
[IMP] The items in the wizard could have old items and should be cleaned.

Commit:
Merge pull request #79 from vauxoo-dev/8.0-vx#4357-osval

[REF] crm_rma_lot_mass_return: change i18n translations and current c…

Commit:
[FIX] crm_claim_rma: fix Vauxoo/rma#81 missing record name

Commit:
[REF] crm_claim_rma: Validate record name

Commit:
Merge pull request #82 from vauxoo-dev/8.0-fix-#81-osval

[FIX] crm_claim_rma: fix Vauxoo/rma#81 missing record name

Commit:
[FIX] Test invoice is paid for assign invoice_line_id to all lots

Commit:
Merge pull request #83 from vauxoo-dev/8.0-fix_invoice_is_paid_for_assign_invoice_line_to_lot-dev-yani

[FIX] Test invoice is paid for assign invoice_line_id to all lots

Commit:
Merge pull request #80 from vauxoo-dev/8.0-clean_items_in_wizard_because_has_trash-dev-yani

[IMP] The items in the wizard could have old items and should be cleaned

Commit:
[FIX] crm_rma_lot_mass_return: add filter when looking for lots

Commit:
[REF] crm_rma_lot_mass_return: coverage

Commit:
[REF] crm_rma_lot_mass_return: remove unused variable

Commit:
Merge pull request #86 from vauxoo-dev/8.0-fix#84-osval

[WIP][FIX] crm_rma_lot_mass_return: add filter when looking for lots

Commit:
[IMP] crm_rma_stock_location: It contains crm_rma_advance_location module into, it has loss and refurbished locations as internal usage

Commit:
[REF] crm_rma_location: Change RMA location's usage from internal -> customer

Commit:
[REF] rma: remove crm_rma_advance_location from dependencies

Commit:
[FIX] crm_rma_location: revert changes for RMA Location usage from 'customer' -> 'internal' again

Commit:
[FIX] crm_rma_stock_location: fix W8103

moylop260 committed:
Merge pull request #89 from vauxoo-dev/8.0-fusion-module-crm_rma_advance_location-into-crm_rma_stock_location-osval

[IMP] crm_rma_stock_location gets contained crm_rma_advance_location

Commit:
[REF] method to create supplier rma returns id list now, and render make all about to render supplier rma

moylop260 committed:
Merge pull request #92 from vauxoo-dev/8.0-refactor_method_render_and_create_supplier_rma-dev-yani

[REF] method to create supplier rma returns ids list now, and render m…

Commit:
[IMP] change date of warranty

nhomar committed:
Merge pull request #93 from vauxoo-dev/8.0-change_date_of_invoice_for_warranty-dev-yani

[IMP] change date of warrant

nhomar commented:
@osvalr can you update the status of this PR to relate it with all the work for rebase?

Commit:
[ADD] deadline is a field of odoo core, it is added to here for add it a help and then make it a functional field. It is added a menu for rma settings for calculate priority and deadline on claim
[ADD] depends unnecessary, configuration fields in res.company equals to rma.settings.
[ADD] priority field and deadline field are compute using configuration parameters in company, date and date_deadline are required, constrains for priority and limit days
[FIX] the invoice date is the field used to calculate priority.
[FIX] add demo data for the test
[FIX] priority must be greater zero in company and rma settings
[FIX] date_deadline is required field and when the supplier claim is created from customer claim, the date_deadline is needed
[ADD] date and date_deadline added to claims for consistency in the demo
[FIX] if limit_days is not configured, should be used a number (1 in this case) for calculate deadline, the constraint needed date and date_deadline for tests constraint date_deadline < date, the deadline like required=True causing the error (WARNING rma openerp.models.schema: Table crm_claim: unable to set a NOT NULL constraint on column date_deadline !) because demo data was added before add required to deadline and deadline in demo data is null. Then required=True is delete
[ADD] manager users can edit deadline field
[ADD] documentation on the methods for calculating deadline
[IMP] order in tree view of claim line
[IMP] deadline added in claim.line, if deadline is major than today, the claim line is red in the tree view

Commit:
[ADD] Portal view My Claims is added just for portal group; demo claim for portal user is added, create rma category for rma groups readonly, user and manager; the users of sale can read claims; RMA User and RMA Manager can see buttons in rma documents

Commit:
[ADD] menuitems that can see the readonly user, stage and claim_type read acl for all user groups, menu stock rma for all users too

nhomar committed:
Merge pull request #96 from vauxoo-dev/8.0-create_rma_groups_and_edit_portal

[ADD] Portal view My Claims is added just for portal group; demo clai…

nhomar committed:
Merge pull request #95 from vauxoo-dev/8.0-compute_deadline_field_and_rma_setting_menu-dev-yani

[ADD] deadline is a field of odoo core, it is added to here for add i…

Commit:
[IMP] Moved the demo users rma_support and rma_manager to crm_claim_rma module
[IMP] Change the user_id field value in demo data of the claims.
[IMP] onchange method added when the user_id or company_id in the claim are change, then, the warehouse_id in the claim change too. Add rma_warehouse_id field in res.users and res.company for fill warehouse in claim.
[IMP] Added two invoices for create demo date in claims about products with very high, high or normal priority, in addition to, two products in red color on tree view because deadline is expired
[ADD] name to the group sale and purcha in rma.config.settings, documentation to limit_days
[ADD] rma_warehouse_id of company is added to rma.config.settings
[FIX] tests about rma configuration is fixed, because the data send to wizard must be initialize, then, the method get_values is requested and then, is modified
[FIX] the name of field is rma_warehouse_id

Commit:
Merge pull request #99 from vauxoo-dev/8.0-improvements_demo_data_and_warehouse_id_added-dev-yani

[IMP] Moved the demo users rma_support and rma_manager to crm_claim_r…

Commit:
[FIX] stock: prevent transfer of picking

Commit:
Merge pull request #101 from vauxoo-dev/8.0-fix_demo_picking_transfer-dev-yani

[FIX] stock: prevent transfer of picking

Commit:
[IMP] It is implied that the subject is of the claim, therefore, claim word is erased.

Commit:
Merge pull request #103 from vauxoo-dev/8.0-delete_claim_word_of_field_label-dev-yani

[IMP] It is implied that the subject is of the claim, therefore, claim word is erased.

Commit:
[ADD] Add supplier invoice to claim line without serial number

Commit:
Merge pull request #104 from vauxoo-dev/8.0-add_supplier_invoice_without_serial_number_dev-yani

[ADD] Add supplier invoice and supplier in the claim line without serial number

Commit:
[FIX] pylint errors

Commit:
Merge pull request #105 from vauxoo-dev/8.0-fix_new_pylint_erros-dev-yani

[FIX] pylint errors

Commit:
[FIX] If the product has prodlot, the invoice_line_id of supplier claim line is get from prodlot, else the invoice_line_id is get from supplier_invoice_id calculated in claim.line

Commit:
Merge pull request #106 from vauxoo-dev/8.0-get_invoice_line_id_when_supplier_claim_is_created-dev-yani

[FIX] If the product has prodlot, the invoice_line_id is get from prodlot, else the invoice_line_get is get from supplier_invoice_id calculated in claim.line

Commit:
[ADD] Description to humans in demo claim lines

Commit:
Merge pull request #107 from vauxoo-dev/8.0-change_description_claim_line-dev-yani

[ADD] Description to humans in demo claim lines

Commit:
[IMP] Tranfers missing product in demo sale order and the quant of prodcut remains in Customers Location because in RMA, when the claim is registrered, the product should be in Customers Location

Commit:
Merge pull request #108 from vauxoo-dev/8.0-transfer_missing_product_in_sale-dev-yani

[IMP] Tranfers missing product in demo sale order

Commit:
[FIX] error when there is a new claim type and it is not customer or supplier.

Commit:
Merge pull request #110 from vauxoo-dev/8.0-fix_when_claim_type_is_not_customer_or_supplier-dev-yani

[FIX] error when there is a new claim type and it is not customer or supplier

Commit:
[FIX] when a supplier claim line is created from customer claim line, the quantity in the customer claim line is not copied into supplier claim line

Commit:
Merge pull request #111 from vauxoo-dev/8.0-pass_quantity_of_claim_line_to_supplier_claim_line-dev-yani

[FIX] when a supplier claim line is created from customer claim line, the quantity in the customer claim line is not copied into supplier claim line

Commit:
[IMP] Method to search the last supplier invoice to get required information using account.invoice.report (#117)

* [IMP] When the product has not serial/lot number, the system has not way to determine the supplier and supplier invoice of product. A new method is added to modularize and search the last supplier invoice  to get required information.

* [ADD] One test with lot and other test without  lot

* [IMP] refactor, to use search instead of SQL

* [FIX] order desc to date_invoice and id

* [IMP] Using bypass-orm sql but now it is fixed. When the product has not serial/lot number, the system has not way to determine the supplier and supplier invoice of product. A new method is added to modularize and search the last supplier invoice to get required information.

* [IMP] to use account.invoice.report to search the supplier invoice

* [IMP] best name to test and id include downward

* [ADD] Commercial partner id added instead of partner id

Commit:
[FIX] W7908(missing-newline-extrafiles), i18n/es_MX.po and i18n/es_VE.po Missing newline

Commit:
Merge pull request #118 from vauxoo-dev/8.0-fix_issue_116_W7908-dev-yani

[FIX] W7908(missing-newline-extrafiles), i18n/es_MX.po and i18n/es_VE.po Missing newline

Commit:
[IMP] improvements in depends

Commit:
Merge pull request #120 from vauxoo-dev/8.0-improvement_in_depends-dev-yani

[IMP] improvements in depends

Commit:
[FIX] refactor in method to fix R0101(too-many-nested-blocks) (#119)

* [FIX] refactor in method to fix R0101(too-many-nested-blocks)

* [FIX] 0199(docstring-first-line-empty), StockProductionLot.default_get] First line empty in method docstring

* [FIX] [C0411(wrong-import-order), ] standard import "from datetime import date" comes before "from openerp.tests.common import TransactionCase"

* [ADD] A purchase with 9 products with different serial number, each will have a supplier invoice line from invoice

Commit:
[IMP] refactor in method of crm_rma_prodlot_invoice module to fix R0101(too-many-nested-blocks) (#121)

* [IMP] refactor in method of crm_rma_prodlot_invoice module to fix R0101(too-many-nested-blocks)

* [FIX] [C0199(docstring-first-line-empty) fixed

* [FIX] [C0411(wrong-import-order), ] standard import "from datetime import date" comes before "from openerp.tests.common import TransactionCase"

* [IMP] Refactor in method invoice_line_create in sale_order_line

* [IMP] one method to set invoice line field in the serial/lot number, and refactor of method in stock_transfer

* [IMP] change name to methods because the code was modularize

* [IMP] redundant condition

* [FIX] take all picking_ids from sale order

* [IMP] add documentation to new methods

* [ADD] test when the invoice type is not exists

* [ADD] test added when sale order have picking invoicing type

Commit:
[IMP] crm_claim_rma: include spanish terms (#122)

[IMP] crm_claim_rma: include spanish terms

Commit:
[IMP] crm_rma_advance_warranty: supplier info is computed here.

i.e: Product A can have two potential suppliers: China Export and Samsung.
But, this information (seller_ids) does not ensure which will be the correct 
supplier of a specific product, by example, a Product A may have been purchased
to the China Export or Samsung supplier.

We need the supplier of the specific damaged product, in this case is supplier_id
in claim_line, and we need a computed field that can call other generic method 
that returns the product.supplierinfo of that product.

Extract part of the logic to get the supplier info for a claim line
- Extract logic "find supplier information for a product" into a
  method to be reused later on.

merge (#123)

Commit:
[FIX] comment the call to do_detailed_transfer because 

when a instance is created using the next steps: 
1. createdb test_bd, 
2. python odoo.py ... -u all, 
3. python odoo.py .... -i rma, 

with addons-vauxoo, then, occur the next error

"You cannot transfer the current picking, because the invoice is already registered with the picking" in addons-vauxoo/stock_picking_invoice_validation/model/stock_transfer_details.py file (#124)

Commit:
[FIX] comment the call to do_detailed_transfer because when a instance is created using the next steps: 1. createdb test_bd, 2. python odoo.py ... -u all, 3. python odoo.py .... -i rma, with addons-vauxoo, then, occur the next error "You cannot transfer the current picking, because the invoice is already registered with the picking" in addons-vauxoo/stock_picking_invoice_validation/model/stock_transfer_details.py file (#124)

Commit:
[FIX] crm_rma_stock_location: update i18n terms, closes #109 (#125)

Commit:
[IMP] Validate that the code has no value and then, to create a new code to the claim. Because there are inheritance of the copy method where depending on the claim type and then the code is assigned, and that value is overwriting in this method.

Commit:
[IMP] crm_rma_claim_make_claim: extract logic to search customer claim lines related (#126)

Commit:
[IMP] When the method copy is called, i.e a claim is duplicated, the code is not set correctly, the copy method should be call to sequence method for obtains the code for the claim (#128)

* [IMP] When the method copy is called, i.e a claim is duplicated, the code is not set correctly, the copy method should be call to sequence method for obtains the code for the claim

* [IMP] tests add to copy method

* [FIX] flake8 in test and model

Commit:
[IMP] Returns claim lines for better user of inheritance (#129)

nhomar committed:
[IMP] crm_rma_lot_mass_return: redesigned the ui of (#94)

* [IMP] redesigned the ui.
* [CC] crm_rma_lot_mass_return: clima_line
* [REF] crm_rma_lot_mass_return: reorganize and rename components
* [REF] crm_rma_lot_mass_return: update i18n
* [FIX] crm_rma_lot_mass_return: C0199
* [FIX] crm_rma_lot_mass_return: W7911
* [IMP] crm_rma_lot_mass_return: include wizard's help

Commit:
[REF] crm_rma_lot_mass_return: include spanish terms for wizard (#131)

Commit:
[IMP] too many nested block in method of crm_rma_lot_mass_return (#130)

* [FIX] crm_rma_lot_mass_return: R0101(too-many-nested-blocks)

 - Solve R0101(too-many-nested-blocks)
 - Solve McCabe rating with 27
 - Solve docstring issues in different models and tests

* [IMP] crm_rma_lot_mass_return: enhance method that parse input_data

 - rewritten get_data_of_product method
 - include isolated unit tests

Commit:
[FIX] crm_rma_lot_mass_return: move picking transfer to tests (#134)

- Uncomment unit tests, this closes #124, #127
 - Transferring picking for sale & purchase orders have
 been move to tests
 - Create a common class for unit tests to avoid code
 duplication later on
 - Make transfer in each test case

nhomar committed:
[FIX] crm_rma_lot_mass_return: properly name assets.

nhomar committed:
[FIX] crm_rma_lot_mass_return: fixes Vauxoo/yoytec#1357

nhomar committed:
[FIX] crm_claim_rma: xml_id is unreachable.

This module one of its features is build the Claims Window
for portal users which convert into a proper solution just add
the dependency properly.

As portal is autoinstall=True then it is always installed
and xml is reachable (like web and/or base), but it is a good practice
set it as dependency.

The xml_id 'portal.portal_after_sales' is unreachable.
The xml_id 'portal_claim.portal_after_sales_claims' is unreachable.
The xml_id 'portal.partner_demo_portal' is unreachable.
The xml_id 'portal_claim.portal_after_sales_claims' is unreachable.

Fixes Vauxoo/yoytec#1352

In Customer App:

The xml_id 'portal.partner_demo_portal' is unreachable

Fixes Vauxoo/yoytec#1414

nhomar committed:
Merge pull request #135 from vauxoo-dev/8.0-fix-undefined

Commit:
[REF] crm_rma_prodlot_supplier: delete default_get method, closes #132 (#133)

- When a serial number is created from picking transfer wizard,
 the supplier is set when the transfer is done, so make
 this method here unnecessary
 - change unit test to adapt new change for default_get method

Commit:
[REF] crm_rma_lot_mass_return: remove one of key actions trigger in js

Having two triggers when a key is pressed and then released
for the same actions it may lead to undesired behavior

Commit:
[FIX] crm_rma_lot_mass_return: correct behavior for wizard

 - Correct behavior based on old algorythm
 - Include unit test

Commit:
[IMP] crm_rma_lot_mass_return: move search lot from invoices to an external method

nhomar committed:
Merge pull request #137 from vauxoo-dev/8.0-fix-red-message-box-osval

[Asking for review][8.0][crm_rma_lot_mass_return] Fix issues in wizard

Commit:
[ADD] Change field type to name field from Char to Text, because needs (#138)

more spacing to describe the problem of product

Commit:
[ADD] Added crm_claim_code with all history

Commit:
Merge branch '8.0' of github.com:Vauxoo/rma into 8.0-crm_claim_separate

nhomar committed:
Merge pull request #140 from vauxoo-dev/8.0-crm_claim_separate

[8.0][crm_claim_code] Added new module to fix the tree dependency

Commit:
[ADD] Added new module crm_claim_type with all history

Commit:
Merge branch '8.0' of github.com:Vauxoo/rma into 8.0-rma-claim_type

nhomar committed:
Merge pull request #141 from vauxoo-dev/8.0-rma-claim_type

[8.0][rma claim type] Added new module crm_claim_type to fix the dependency tree

Commit:
[DEL] bye the crm_claim_code module. The contains of crm_claim_code module is migrated to crm_claim_rma module

Commit:
[DEL] bye the crm_claim_type module. The contains of crm_claim_type module is migrated to crm_claim_rma module

Commit:
[IMP] crm_claim_rma: crm_claim_code, crm_claim_type and crm_claim_rma_code are unified.
[FIX] The author attribute in __openerp__.py was too long. The error was
DataError: value too long for type character varying(128)
[IMP] To use the warehouse_id field from default_warehouse_from_sale_team
module for that warehouse_id field will be take the default_warehouse
from sales team of user.
[FIX] context for claim_type field is moved from crm_claim.xml to
crm_claim_type_from_type.xml

Commit:
[DEL] bye the crm_claim_rma_code module. The contains of crm_claim_rma_code module is migrated to crm_claim_rma module
[IMP] travis.yml: THe Vauxoo/addons-vauxoo is added to travis.yml
because the default_warehouse_from_sale_team module needs to be
installed.
[FIX] the pre_init_hook and post_init_hook is moved to crm_claim_rma,
these methods create a code to existing claims, because the code field
is a required field now.

Commit:
[IMP] crm_claim_rma: Uniting the views concerning crm.claim
[IMP] Uniting the security in just one.
[DEL] unnecessary sequences are deleted
[IMP] arrange fields in crm_claim and claim_line
[IMP] arrange files in __openerp__.py and remove from_* in the files
migrated
[FIX] flake and pylint fixed

Commit:
[IMP] Bye to crm_rma_stock_location module: crm_rma_stock_location module was joined woth crm_claim_rma
[ADD] the post_init_hook for create_locations_rma from
crm_rma_stock_location is added to old post_init_hook

Commit:
[IMP] Bye to crm_rma_lot_mass_return module: crm_rma_lot_mass_return module was joined with crm_claim_rma

Commit:
[IMP] Bye to crm_rma_prodlot_supplier module: crm_rma_prodlot_supplier was joined with crm_rma_prodlot_invoice
[IMP] do_detailed_transfer method from crm_rma_prodlot_supplier module is
refactored to best method. It is joined with do_detailed_transfer from
crm_rma_prodlot_invoice module

Commit:
[ADD] stock_unfuck is added to crm_claim_rma module to can force quants when a picking is assign

Commit:
[FIX] crm_claim_rma: - Documentation added in _get_warranty_limit_values method
- Raises in set_warranty_limit now it is used directly in _get_warranty_limit_values
- auto_set_warranty method is deleted because already is not used in the view, it was used by a button
- The exceptions now are inheriting from the library UserError
- Add a raises when a product have more than one supplier activated

Commit:
[IMP] Improved return in method for calculate limit days.
- Add a test in test_rma_configuration when the user rma try write a date_deadline manually

Commit:
[RMA] refactor of RMA modules (based on 8.0-puchase) (#1)

* [REF] crm_claim_rma: refactor module to enhance readability and coverage

 - try refunding a refunded claim
 - validate return_value
 - validate invoice date exception is thrown
 - add new unit test for supplier claims
 - include tests for supplier claims and warranty durations
 - simplify how warning message get computed based on claim date and limit date
 - validate missing field values exceptions, and include test_contrains in unit tests
 - test a claim without invoice date
 - remove commented import
 - include missing test
 - change claim_id
 - remove unused method
 - warehouse and location there will be always be at least one and its rma location is created automatically
 - test _default_description
 - include use case when user tries to re-create a product return
 - refactor code for picking wizard
 - minimize conditionals when computing picking_type and field to be written out
 - include test case for challenge claim line copy
 - include test case for multiple addres when creating a product return, and also change exception use in product picking in claim wizard
 - test when partner location for each line is different, and also when the return address (in claim lines) are different
 - include constrain validation when limit days in company records reaches zero or less
 - if (after an if) -> elif in order to avoid asking
 - rename var wizard_id -> line_id,
 - include lot mass return test 2, it was missing from __init__
 - better readability for method
 - include test case for add-lines wizard for supplier claim
 - refactor search method to find lot by name
 - refactor get_lot_ids method
 - include unit test case when warning and empty values are expected when adding claim lines
 - simplify conditions
 - delete method change_list, it seems is not used
 - include help buttons test case

* [CC] crm_rma_advance_warranty

* [REF] crm_rma_advance_warranty: include test cases

[REF] crm_rma_advance_warranty: include test case for supplier warranty lot-based

* [REF] crm_rma_advance_warranty: fix test case

* [REF] crm_claim_rma: remove method unused _get_stock_moves_with_code

* [REF] crm_claim_rma: remove dependency, is already on rma app

* [DEL] crm_rma_prodlot_invoice: remove unused field

* [REF] crm_claim_rma: call action_cancel method

* [REF] crm_rma_claim_make_claim: refactor for better read

* Revert "[REF] crm_rma_claim_make_claim: refactor for better read"

This reverts commit 0a2015f63059f0f64fa5aa4f9adb3b136916f080.

* [REF] crm_claim_rma: include test for mail messages

* [REF] crm_claim_rma: fix lints

* [CC] crm_rma_advance_warranty

* [CC] crm_rma_advance_warranty

* [REF] crm_rma_prodlot_invoice: remove redundant line

* [REF] crm_rma_advance_warranty: include claim line warranty test use cases

* [REF] crm_claim_rma: validate mail tests

* [REF] crm_claim_rma: rename compute methods

Commit:
[REF] crm_claim_rma: remove sale order demo

 - create customer
 - create sale order in unit test instead xml data

Commit:
[REF] crm_claim_rma: remove unused method from tests and remove base.res_partner references from test_picking creation

Commit:
[REF] crm_claim_rma: use test-created partner

Commit:
[FIX] crm_claim_rma: fix lints

Commit:
[REF] crm_claim_rma: rename lot_mass_return_tests_common -> common

Commit:
[REF] crm_claim_rma: remove external sale demo from unit tests

Commit:
[FIX] crm_claim_rma: create a customer in yaml to be used in tests with unlimited credit limit

Commit:
[FIX] crm_claim_rma: fix lints

Commit:
[REF] crm_claim_rma: refactor

Commit:
[REF] crm_claim_rma: use new partner for invoice in demo

Commit:
[FIX] crm_claim_product_supplier: create a customer to be used in tests with unlimited credit limit

Commit:
Merge pull request #2 from vauxoo-dev/8.0-remove-sale-order-demo-osval

Commit:
[REF] crm_claim_rma: delete repeated code

Commit:
[REF] crm_claim_rma: refactor unit tests

Commit:
[REF] crm_claim_rma: dummy refactors made

Commit:
[FIX] crm_claim_rma: include method change_list back

Commit:
[IMP] missing assets.xml in __openerp__.py about functionalities of magic wizard (# 4)

Commit:
[FIX] When a supplier claim is created from customer claim, the initial stage of supplier claim should be the default stage in that moment, because the initial stage of supplier claim can be changed in the configuration of stages (# 5)

Commit:
[FIX] When a supplier claim is created from customer claim, the initial stage of supplier claim should be the default stage in that moment, because the initial stage of supplier claim can be changed in the configuration of stages (# 6)

nhomar committed:
Merge pull request #142 from vauxoo-dev/8.0-purchase

[DEL] bye the crm_claim_code module. The contains of crm_claim_code module is migrated to crm_claim_rma module

Commit:
[REF] crm_claim_rma: remove "xrange built-in referenced" lint using another approach in test case

Commit:
[REF] crm_rma_prodlot_invoice: remove "xrange built-in referenced" lint using another approach in test case

Commit:
[FIX] crm_claim_rma: E302, E501, C0112

Commit:
[ADD] crm_claim_rma: Add demo readonly user for tests (#146)

nhomar committed:
Merge pull request #145 from vauxoo-dev/8.0-purchase-fix-lints-osval

[WIP][8.0-purchase] Fix lints and others errors

Commit:
[FIX] crm_claim_rma: claim line name_get wasn't rendering record sets (#148)

Commit:
[8.0-purchase] move exceptions osval (#147)

* [REF] crm_claim_rma: move custom exceptions to centralized file in module's root dir

* [REF] crm_claim_rma: include new exception for supplier warranty duration

 - when a product warranty period depends on how much time frame the supplier
had considered for it. In case is not set for the supplier's product when trying to compute
products warranty duration, the user needs to know what's happening

nhomar committed:
Merge pull request #149 from Vauxoo/8.0-purchase

8.0 purchase

Commit:
[FIX] crm_claim_rma: The Customer Support user can not validate the claim (#150)

[FIX] crm_claim_rma: The Customer Support user can not validate the claim because because the user lacks permissions on some models of stock.

Commit:
[FIX] crm_claim_rma: fix python and javascript lints (#152)

- C0411, C8107,  E8103,  C8108
 - delete invoice_no_date from models directory
 - remove ProductWithoutSupplierWarranty exception, because it is
 being used only once
 - W0235(useless-super-delegation)
 - hide groups and create new ones instead of replacing them for
 W7940(dangerous-view-replace-wo-priority)
 - remove duplicate xml ids: W7902(duplicate-xml-record-id)
 - Fix some W7903(javascript-lint)
 - Disable W0235 for exceptions

Commit:
[REF] crm_claim_rma: enable lint, remove redundant field label (#154)

Commit:
[REF] crm_rma_prodlot_invoice: change the way the invoice line related to the lot is checked in sale order' invoice lines (#155)

Commit:
[REF] Update translations and fix lints (#157)

Commit:
[IMP] crm_claim_rma: New method for obtain the products that will be calculate in rma stock (#158)

moylop260 committed:
[FIX] seudo fix performance issue (#160)

Commit:
[Waiting for merge][8.0] Allow edit and change manually priority for a product (#159)

[8.0] Allow edit and change manually priority for a product

Commit:
[FIX] crm_claim_rma: Remove fields from product variant\'s view because there is no case to use in there, The real use of these fields can be found in RMA \> Customer Stock RMA

Commit:
[FIX] crm_rma_advance_warranty: Fix flaky test, ensure that the warranty is not_define (#163)

Commit:
[ADD] crm_rma_location: Create new picking for destruction of products come from RMA to Loss location (#164)

* [ADD] crm_rma_location: Create new picking for destruction of products come from RMA to Loss location

* [IMP] crm_claim_rma and crm_rma_location: Move location related with rma from crm_claim_rma to crm_rma_location because the crm_rma_location module has everything related to stock

Commit:
[IMP] yoytec_rma_workflow: Add destruction picking type to warehouse view (#165)

Commit:
[REF] crm_rma_claim_make_claim: include log message when product have been added to claim (#162)

nhomar commented:
This PR is outdated and can be closed.
